### DISCUSSION TITLE
Access to $store

### DISCUSSION THREAD

**Discussion by kisscodes1909** [alpinejs/alpine#1469] [7. Bugs (archived)]
I used Spruce for my project, I checked Alpine yesterday and I saw alpine has supported $store.

Old syntax with Spruce
Spruce.store('user', {
    imageProfile: siteGeneralSetting.defaultBuyerProfileImg,
    email: '',
    userName: '',
});

Before I used Spruce.store('user').email to access but now when I declare

New syntax with Alpine

document.addEventListener('alpine:initializing', () => {
    Alpine.store('user, {
        imageProfile: siteGeneralSetting.defaultBuyerProfileImg,
        email: '',
        userName: '',
    })
})

How to access to Alpine store ?

I tried Alpine.store('user').email but I got Alpine.store is not a function

![screenshot-alpinejs dev-2021 06 14-09_35_19](https://user-images.githubusercontent.com/21024998/121832578-9511f280-ccf4-11eb-946b-40b0ee5042f8.png)

**Comment by SimoTod** [comment]
It would be `$store.user.email` (https://alpinejs.dev/magics/store)
Note, you have a typo as well, you called your store `darkMode`

**Reply by kisscodes1909** [reply]
Hi @SimoTod, Thank you for your response

the problem is I want to access it from after declaration, not from the template

Please check my code:

`
Spruce.store('user', {
    imageProfile: siteGeneralSetting.defaultBuyerProfileImg,
    email: '',
    userName: '',
});


firebase.auth().onAuthStateChanged(function(user) { 
    FBGetUserMeta(user).then((userMeta)=>{

        if( user.email ) {
            Spruce.store('user').email = user.email;
        }

        if( user.displayName) {
            Spruce.store('user').userName = user.displayName;
        }

        if( !userMeta.exist ) return;

        const userMetaData = userMeta.doc();

        if( userMetaData.imageProfile ) {
            Spruce.store('user').imageProfile = userMetaData.imageProfile;
        }

    })

    // Attach FBUpdateLastActivity interval 1 min
    setInterval(() => {
        FBUpdateLastActivity(user).then(()=> {
        })
    }, 1000);

});

`

Can I use Alpine.store('user') with the latest version from Alpine?

I'm using "alpinejs": "^2.8.2"

**Reply by SimoTod** [reply]
Store is only supported in Alpine v3 (currently in beta).

About your code, I would build a user object and just call Alpine.store('user' , yourObject) at the end (but again, you need v3)

**Reply by ryangjchandler** [reply]
`Alpine.store` without the second argument does actually return the `Proxy` object in v3 - you need to make sure you call it after you've initialised the store.

**Reply by kisscodes1909** [reply]
Thank @SimoTod, @ryangjchandler 

also, I see a problem, if I initialize big spruce the UI will lag, will it be fixed in v3?

**Reply by ryangjchandler** [reply]
@kisscodes1909 You'll have to try it out - I haven't upgraded any of my larger apps to v3 yet as I'm waiting for it to be completely stable.
